flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/coordination/ClientCoordinationHandler.java

```diff
@@ -77,21 +77,29 @@ protected CompletableFuture<ClientCoordinationResponseBody> handleRequest(
         OperatorID operatorId = request.getPathParameter(OperatorIDPathParameter.class);
         SerializedValue<CoordinationRequest> serializedRequest =
                 request.getRequestBody().getSerializedCoordinationRequest();
-        CompletableFuture<CoordinationResponse> responseFuture =
-                gateway.deliverCoordinationRequestToCoordinator(
-                        jobId, operatorId, serializedRequest, timeout);
-        return responseFuture.thenApply(
-                coordinationResponse -> {
-                    try {
-                        return new ClientCoordinationResponseBody(
-                                new SerializedValue<>(coordinationResponse));
-                    } catch (IOException e) {
-                        throw new CompletionException(
-                                new RestHandlerException(
-                                        "Failed to serialize coordination response",
-                                        HttpResponseStatus.INTERNAL_SERVER_ERROR,
-                                        e));
-                    }
-                });
+        try {
+            CompletableFuture<CoordinationResponse> responseFuture =
+                    gateway.deliverCoordinationRequestToCoordinator(
+                            jobId, operatorId, serializedRequest, timeout);
+            return responseFuture.thenApply(
+                    coordinationResponse -> {
+                        try {
+                            return new ClientCoordinationResponseBody(
+                                    new SerializedValue<>(coordinationResponse));
+                        } catch (IOException e) {
+                            throw new CompletionException(
+                                    new RestHandlerException(
+                                            "Failed to serialize coordination response",
+                                            HttpResponseStatus.INTERNAL_SERVER_ERROR,
+                                            e));
+                        }
+                    });
+        } catch (Exception e) {
+            throw new RestHandlerException(
```

Comment:
Who is throwing an exception here? Errors shoud be returned via the returned future; they should never fail immediately.